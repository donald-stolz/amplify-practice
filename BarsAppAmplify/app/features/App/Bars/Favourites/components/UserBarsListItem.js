// @flow
import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  ActivityIndicator,
  Dimensions,
} from 'react-native';
import moment from 'moment';
import Swipeout from 'react-native-swipeout';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Foundation from 'react-native-vector-icons/Foundation';
import MapLinks from '../../../../../components/MapLinks';

// Config
import * as COLORS from '../../../../../config/colors';

// Types
type Props = {
  item: {
    id: string,
    name: string,
    phone: string,
    location: string,
    website: string,
    lat: string,
    lng: string,
    createdAt: string,
  },
  deleteFavourite: Function,
  openWebsiteLink: Function,
  toggleMapLinks: Function,
  openPhone: Function,
  isVisible: boolean,
  deleting: boolean,
};

const { width } = Dimensions.get('window');

const UserBarsListItem = ({
  item,
  deleteFavourite,
  openWebsiteLink,
  toggleMapLinks,
  openPhone,
  isVisible,
  deleting,
}: Props) => {
  const swipeoutBtns = [
    {
      backgroundColor: deleting ? COLORS.DEFAULT_PRIMARY_COLOR : COLORS.ACCENT_COLOR,
      onPress: () => deleteFavourite(item.id),
      text: deleting ? <ActivityIndicator color={COLORS.TEXT_PRIMARY_COLOR} /> : 'DELETE',
    },
  ];
  const date = moment.utc(item.createdAt).format('MMMM Do YYYY, h:mm:ss a');
  return (
    <Swipeout right={swipeoutBtns} backgroundColor={COLORS.TEXT_PRIMARY_COLOR}>
      <View style={styles.card}>
        <View style={styles.details}>
          <Text style={styles.header}>{item.name}</Text>
          <Text style={styles.location}>{item.location}</Text>
          <Text style={styles.phone}>{item.phone}</Text>
          <Text style={styles.date}>{`Added on ${date}`}</Text>
        </View>
        <View style={styles.iconWrapper}>
          <TouchableOpacity onPress={() => openWebsiteLink(item.website)}>
            <MaterialCommunityIcons name="web" size={18} color={COLORS.ACCENT_COLOR} />
          </TouchableOpacity>
          <TouchableOpacity onPress={toggleMapLinks}>
            <MaterialCommunityIcons name="directions" size={18} color={COLORS.DARK_PRIMARY_COLOR} />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => openPhone(item.phone)}>
            <Foundation name="telephone" size={18} color={COLORS.PRIMARY_TEXT_COLOR} />
          </TouchableOpacity>
        </View>
        <MapLinks
          isVisible={isVisible}
          onCancelPressed={toggleMapLinks}
          onAppPressed={toggleMapLinks}
          onBackButtonPressed={toggleMapLinks}
          name={item.name}
          lat={parseFloat(item.lat)}
          lng={parseFloat(item.lng)}
          barId={item.id}
        />
      </View>
    </Swipeout>
  );
};

const styles = StyleSheet.create({
  card: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: 10,
    paddingHorizontal: 20,
    paddingVertical: 10,
    width,
  },
  details: {
    width: '90%',
  },
  header: {
    fontSize: 18,
    fontWeight: '500',
  },
  location: {
    fontSize: 14,
  },
  phone: {
    fontSize: 10,
  },
  date: {
    fontSize: 10,
    color: COLORS.SECONDARY_TEXT_COLOR,
  },
  iconWrapper: {
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});

export default UserBarsListItem;
