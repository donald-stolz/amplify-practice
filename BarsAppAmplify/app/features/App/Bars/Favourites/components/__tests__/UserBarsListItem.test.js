import React from 'react';
import { shallow } from 'enzyme';
import UserBarsListItem from '../UserBarsListItem';

describe('UserBarsListItem', () => {
  it('Renders correctly', () => {
    Date.now = jest.fn(() => new Date(Date.UTC(2017, 7, 9, 8)).valueOf());
    const deleteFavourite = jest.fn();
    const openWebsiteLink = jest.fn();
    const toggleMapLinks = jest.fn();
    const openPhone = jest.fn();
    const wrapper = shallow(
      <UserBarsListItem
        item={{ id: '123' }}
        deleteFavourite={deleteFavourite}
        openWebsiteLink={openWebsiteLink}
        toggleMapLinks={toggleMapLinks}
        openPhone={openPhone}
        isVisible
        deleting={false}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
