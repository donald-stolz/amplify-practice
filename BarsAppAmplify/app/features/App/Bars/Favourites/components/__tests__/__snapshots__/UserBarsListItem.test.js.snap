// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBarsListItem Renders correctly 1`] = `
<Swipeout
  backgroundColor="#FFFFFF"
  disabled={false}
  right={
    Array [
      Object {
        "backgroundColor": "#FF4081",
        "onPress": [Function],
        "text": "DELETE",
      },
    ]
  }
  rowID={-1}
  sectionID={-1}
  sensitivity={50}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 10,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Component
      style={
        Object {
          "width": "90%",
        }
      }
    >
      <Component
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "500",
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 14,
          }
        }
      />
      <Component
        style={
          Object {
            "fontSize": 10,
          }
        }
      />
      <Component
        style={
          Object {
            "color": "#757575",
            "fontSize": 10,
          }
        }
      >
        Added on August 9th 2017, 8:00:00 am
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Icon
          allowFontScaling={false}
          color="#FF4081"
          name="web"
          size={18}
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[MockFunction]}
      >
        <Icon
          allowFontScaling={false}
          color="#0288D1"
          name="directions"
          size={18}
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Icon
          allowFontScaling={false}
          color="#212121"
          name="telephone"
          size={18}
        />
      </TouchableOpacity>
    </Component>
    <MapLinks
      barId="123"
      isVisible={true}
      lat={NaN}
      lng={NaN}
      onAppPressed={[MockFunction]}
      onBackButtonPressed={[MockFunction]}
      onCancelPressed={[MockFunction]}
    />
  </Component>
</Swipeout>
`;
